blockbuster/src/programs/bubblegum/mod.rs

```diff
@@ -252,3 +251,56 @@ fn build_creator_verification_payload(
         args,
     })
 }
+
+// See Bubblegum for offsets and positions:
+// https://github.com/metaplex-foundation/mpl-bubblegum/blob/main/programs/bubblegum/README.md#-verify_collection-unverify_collection-and-set_and_verify_collection
+// Note: Collection from the args is used to update the collection in Bubblegum, so use it as the source of truth (instead of collection via accounts).
+fn build_collection_verification_payload(ix_data: &[u8]) -> Result<Payload, BlockbusterError> {
+    let metadata_offset = 108;
+    let collection_byte_size = 32;
+    let metadata_byte_end = ix_data.len() - collection_byte_size;
+
+    // Ensure data is valid.
+    if ix_data.len() < metadata_offset + collection_byte_size {
+        return Err(BlockbusterError::InstructionParsingError);
+    }
+
+    let args_raw = ix_data[metadata_offset..metadata_byte_end].to_vec();
+    let args = MetadataArgs::try_from_slice(&args_raw)?;
+    let collection_raw = ix_data[metadata_byte_end..].to_vec();
+    let collection: Pubkey = Pubkey::try_from_slice(&collection_raw)?;
+
+    Ok(Payload::CollectionVerification {
+        collection,
+        args,
+        verify: true,
+    })
+}
+
+// See Bubblegum for offsets and positions:
+// https://github.com/metaplex-foundation/mpl-bubblegum/blob/main/programs/bubblegum/README.md#-verify_collection-unverify_collection-and-set_and_verify_collection
+// NOTE: Unverfication does not include collection. This needs to be fixed in the README.
+fn build_collection_unverification_payload(
+    keys: &[plerkle_serialization::Pubkey],
+    ix_data: &[u8],
+) -> Result<Payload, BlockbusterError> {
+    let collection_index = 8;
+    let metadata_offset = 108;
+    if ix_data.len() < metadata_offset {
+        return Err(BlockbusterError::InstructionParsingError);
+    }
+
+    let args_raw = ix_data[metadata_offset..].to_vec();
+    let args = MetadataArgs::try_from_slice(&args_raw)?;
+    let collection_raw = keys
+        .get(collection_index)
+        .ok_or(BlockbusterError::InstructionParsingError)?
+        .0;
```

Comment:
This is just an FYI, that I also had a draft PR for blockbuster that added explicit support for Verify and Unverify collection: github.com/metaplex-foundation/blockbuster/pull/21.

But I ended up not needing the change for the ordering PR since I added a verify flag and additional sequence number to manage the data out of order.

I don't think this results in a change for what you are doing, I'm just telling you as more of an FYI.

I suppose one thing you could consider is just using index 8 from the accounts to get collection every time, since its always in the same accounts location for `VerifyCollection`, `UnverifyCollection`, and `SetAndVerifyCollection`.

Reply:
Is there a difference between the index 8 account and the one in the instruction args? Why do both exist?